File: src/test/java/com/example/rocko/InvertBinaryTreeTest.java
package com.example.rocko;


import com.example.rocko.lib.TreeNode;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

/**
 * Created by Rocko on 2016/11/30.
 */
public class InvertBinaryTreeTest {

    private InvertBinaryTree invertBinaryTree;
    private TreeNode treeNode;

    @Before
    public void setUp(){
        invertBinaryTree = new InvertBinaryTree();
        treeNode = new TreeNode(4);
        treeNode.left = new TreeNode(2);
        treeNode.left.left = new TreeNode(1);
        treeNode.left.right = new TreeNode(3);

        treeNode.right = new TreeNode(7);
        treeNode.right.left = new TreeNode(6);
        treeNode.right.right = new TreeNode(9);
    }

    @Test
    public void testInvertBinaryTree(){


        TreeNode act = invertBinaryTree.invertTree(treeNode);

        Assert.assertEquals(7,act.left.getVal());
        Assert.assertEquals(2,act.right.getVal());


    }

}

File: tests/src/InvertBinaryTreeTest.php
<?php

namespace Rocko;

use PHPUnit\Framework\TestCase;

/**
 * 左邊跟右邊的元素交換
 */
class InvertBinaryTreeTest extends TestCase
{
    /** @var  InvertBinaryTree */
    protected $invertBinaryTree;
    
    public function setUp()
    {
        $this->invertBinaryTree = new InvertBinaryTree();
    }
    
    /**
     * @group InvertBinaryTreeTest
     * @test
     */
    public function given_2_4_3_should_return_2_3_4()
    {
        //Arrange
        
        //Act
        $actual = $this->invertBinaryTree->invertTree([2, 4, 3]);
        
        //Assert
        $this->assertEquals([2, 3, 4], $actual);
    }
    
    /**
     * @group InvertBinaryTreeTest
     * @test
     */
    public function given_3_5_1_should_return_3_1_5()
    {
        //Arrange
        
        //Act
        $actual = $this->invertBinaryTree->invertTree([3, 5, 1]);
        
        //Assert
        $this->assertEquals([3, 1, 5], $actual);
    }
    
    
    /**
     * @group InvertBinaryTreeTest
     * @test
     */
    public function given_4_2_7_1_3_6_9_should_return_4_7_2_9_6_3_1()
    {
        //Arrange
        
        //Act
        $actual = $this->invertBinaryTree->invertTree([4, 2, 7, 1, 3, 6, 9]);
        
        //Assert
        $this->assertEquals([4, 7, 2, 9, 6, 3, 1], $actual);
    }
}